### PR TITLE
feat(detection): App detects when the user is on campus.

### LINKED ISSUE #154: On campus detection
**Description:**
When the app detects that you are on campus, the red indicator turns to green on the map, and vice-versa.

**Criteria:**

- [x] Localization detection is accurate and in real time.
- [x] Indicator is updated in real time according to localization.

### PR DESCRIPTION
**Pull request by khlilinfo**
### **Summary**
This PR adds EPFL on-campus detection and a corresponding UI indicator to the existing “Study Together” screen. When the user’s location falls inside a bounding box around the EPFL Lausanne campus, the app now shows a pill at the top of the screen with a green dot and the label “On EPFL campus”; otherwise, it shows a red indicator and “Outside of EPFL campus”. The logic lives in the StudyTogetherViewModel, and the feature is fully covered by Compose tests. As part of this, the friends dropdown was visually refined to better align with the new indicator.

### **What’s in this PR**

- State & Logic
         - Adds `isOnCampus`: Boolean to `StudyTogetherUiState`.
         - Adds `isOnEpflCampus(LatLng)` helper in `StudyTogetherViewModel` and wires it into `consumeLocation`.
- UI
         - New `OnCampusIndicator` pill at the top of the Study Together screen (green vs red state).
         - Adjusts the friends pill styling/position to avoid overlap and visually match the indicator.

### **Implementation Notes**

- Campus detection uses a simple EPFL bounding box in the `ViewModel`; UI just reacts to` isOnCampus`.
- Presence and Firestore behavior remain unchanged; on-campus state is purely a client-side UX hint.
- `StudyTogetherScreen(showMap = false)` still renders the indicator so it’s easy to test without Google Maps.

### **Testing**

- New compose test in `StudyTogetherScreenTest and StudyTogetherViewModelTest`
          - Uses FakeFriendRepository and StudyTogetherViewModel.
          - Calls consumeLocation with coordinates inside the EPFL bbox.
- Manual check on emulator/device to confirm indicator toggles correctly and UI layout looks good.

### **Screenshots / Demos**
image
image

### **Checklist**

- [x]  Add isOnCampus to StudyTogetherUiState
- [x]  Implement EPFL bounding-box helper and update consumeLocation
- [x]  Add OnCampusIndicator composable and wire it into the screen
- [x]  Adjust friends pill layout/style to align with the indicator
- [x]  Add Compose test for the on-campus state

**Notes**: 
- Parts of this PR (and some wiring/tests) were generated with AI

### **Issue Reference**

Closes #154

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherViewModel.kt [hunk 1]
```diff
@@ -168,4 +170,19 @@ class StudyTogetherViewModel(
...
+  private fun isOnEpflCampus(position: LatLng): Boolean {
+    val lat = position.latitude
+    val lng = position.longitude
+
+    val minLat = 46.515
+    val maxLat = 46.525
+    val minLng = 6.555
+    val maxLng = 6.575
+
+    println(lat)
```
Comment (Adessrrrr) [inline app/src/main/java/com/android/sample/ui/location/StudyTogetherViewModel.kt]: There is a debugging println here that  should be removed, as the feature works now.

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt [hunk 2]
```diff
@@ -72,6 +74,8 @@ private const val TAG_FIELD_UID = "field_friend_uid"
 private const val TAG_BTN_FRIENDS = "btn_friends"
 private const val TAG_MAP_STUB = "map_stub"
 
+private const val ON_CAMPUS = "on_campus_indicator"
```
Comment (Adessrrrr) [inline app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt:81]: Please avoid hardcoded strings, put them in strings XML instead.

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt [hunk 3]
```diff
@@ -407,40 +427,108 @@ private fun EdumonFriendsDropdown(
...
+                  .padding(top = 44.dp) // show under the pill
+                  .widthIn(min = 220.dp, max = 320.dp),
+          shape = RoundedCornerShape(24.dp),
+          colors =
+              CardDefaults.cardColors(
+                  containerColor = MaterialTheme.colorScheme.surface.copy(alpha = 0.98f)),
+          elevation = CardDefaults.cardElevation(defaultElevation = 6.dp)) {
+            if (friends.isEmpty()) {
+              Text(
+                  text = "No friends yet",
```
Comment (Adessrrrr) [inline app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt:556]: Same here, please avoid hardcoded strings.

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt [hunk 4]
```diff
@@ -407,40 +427,108 @@ private fun EdumonFriendsDropdown(
...
+              containerColor = MaterialTheme.colorScheme.surface.copy(alpha = 0.95f)),
+      elevation = CardDefaults.cardElevation(defaultElevation = 6.dp)) {
+        if (onCampus) {
+          Row(
+              modifier = Modifier.padding(horizontal = 12.dp, vertical = 8.dp),
+              verticalAlignment = Alignment.CenterVertically) {
+                // Green dot
+                Box(modifier = Modifier.size(10.dp).clip(CircleShape).background(StudyGreen))
+                Spacer(Modifier.width(8.dp))
+                Text(text = "On EPFL campus", style = MaterialTheme.typography.labelLarge)
```
Comment (Adessrrrr) [inline app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt:643]: Also same problem here, please avoid hardcoded strings.

**Review by Adessrrrr — CHANGES_REQUESTED** [review]
This PR is a very nice addition and preparation for the next features (focus mode toggling when on campus), it works fine and the cursor updates correctly when we really are at EPFL.
So, the code is good and the feature works, it's a nice addition to the project.

The code is tested well, with a 94.4 % line coverage which is excellent, also there are no issues on Sonar so it's good on this point.

To be merged, you only have to adjust very small details (delete a println and avoid hardcoded strings), then it will be perfect.


Note: An A.I assistant's help has been used to make this review.

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt (hunk 2 above)
Comment (khlilinfo) [reply app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt:81]: this is a testTag

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt (hunk 3 above)
Comment (khlilinfo) [reply app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt:556]: I tried to create a string in string.xml and call it here with R.string.string_name.toSting but this prints the string number and not the acual string i did find a solution

File: app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt (hunk 4 above)
Comment (khlilinfo) [reply app/src/main/java/com/android/sample/ui/location/StudyTogetherScreen.kt:643]: same as above

**Review by Adessrrrr — APPROVED** [review]
Normally all the requested changes have been handled, also it's very good that you added more tests, especially ones that cover edge cases.

I think this can be merged now.
